sputnikdao2/src/proposals.rs

```diff
@@ -635,11 +726,11 @@ impl Contract {
     /// move proposal to "Failed" state.
     #[private]
     pub fn on_proposal_callback(&mut self, proposal_id: u64) -> PromiseOrValue<()> {
-        let mut proposal: Proposal = self
+        let mut proposal: ProposalV1 = self
             .proposals
             .get(&proposal_id)
             .expect("ERR_NO_PROPOSAL")
-            .into();
+            .latest_version();
```

Comment:
Is there some limitation of `Into` that we had to refactor it into a custom `.latest_version` method?